src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,197 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import org.junit.jupiter.api.Test;
+import org.xml.sax.Attributes;
+import org.xml.sax.helpers.DefaultHandler;
+
+public class XdocsUrlTest {
+
+    public static final Path PACKAGE_NAME_PATH =
+            Paths.get("src/main/java/com/puppycrawl/tools/checkstyle/checks");
+
+    public static final String NODE_NAME = "a";
+
+    public static final String SUFFIX_JAVA = ".java";
+
+    public static final String SUFFIX_CHECK_JAVA = "Check.java";
+
+    public static final String MISC = "misc";
+
+    public static final String ANNOTATION = "annotation";
+
+    public static final String COMMENTS_INDENTATION = "CommentsIndentation";
+
+    public static final String INDENTATION = "Indentation";
+
+    public static final String SUPPRESS_WARNINGS_HOLDER = "SuppressWarningsHolder";
+
+    public static final Path AVAILABLE_CHECKS_PATH = Paths.get("src/xdocs/checks.xml");
+
+    private static Map<String, String[]> getXdocsMap() {
+        final File checks = new File(PACKAGE_NAME_PATH.toString());
+        final String[] checksSubDirNames = checks.list((rootDir, pathname) -> {
+            return !pathname.endsWith(SUFFIX_JAVA);
+        });
+        final Map<String, String[]> checksNamesMap = new HashMap<>();
+        if (checksSubDirNames != null) {
+            final FilenameFilter filter = (rootDir, pathname) -> {
+                return pathname.endsWith(SUFFIX_CHECK_JAVA);
+            };
+            for (String dirName : checksSubDirNames) {
+                final File dirFile = new File(PACKAGE_NAME_PATH + "/" + dirName);
+                final String[] fileNames = dirFile.list(filter);
+                checksNamesMap.put(dirName, fileNames);
+            }
+            final String[] miscNames = checks.list(filter);
+            checksNamesMap.put(MISC, miscNames);
+        }
+        return checksNamesMap;
+    }
+
+    @Test
+    public void testXdocsUrl() throws Exception {
+        final SAXParserFactory parserFactory = SAXParserFactory.newInstance();
+        final SAXParser parser = parserFactory.newSAXParser();
+        final CheckTest checkHandler = new CheckTest(NODE_NAME);
+        final List<List<String>> list;
+        try (InputStream input = Files.newInputStream(AVAILABLE_CHECKS_PATH)) {
+            parser.parse(input, checkHandler);
+            list = checkHandler.list;
+        }
+        final Map<String, String[]> checksNamesMap = getXdocsMap();
+        for (List<String> sub : list) {
+            final String moduleName = sub.get(0);
+            final String checkName1 = sub.get(1);
+            final String checkName2 = sub.get(2);
+            final String errorMessage =
+                    "URL in" + AVAILABLE_CHECKS_PATH + " is inconsistent: "
+                            + moduleName + " " + checkName1 + " " + checkName2;
```

Comment:
Please use String.format

Reply:
String.format is forbidden, so I retain it
[ERROR] Forbidden method invocation: java.lang.String#format(java.lang.String,java.lang.Object[]) [Uses default locale]
[ERROR]   in com.puppycrawl.tools.checkstyle.internal.XdocsUrlTest (XdocsUrlTest.java:99)
[ERROR] Scanned 2034 class file(s) for forbidden API invocations (in 1.54s), 1 error(s).

Reply:
It is not forbidden, its usage without explicit locale setting is forbidden.
You should use String.format(Locale.ROOT, "template", args) method
You can find examples in codebase

Reply:
done~